### DISCUSSION TITLE
Adding an arbitrary element through the parsers

### DISCUSSION THREAD

**Discussion by iamhectorsosa** [remarkjs/remark#1105] [Q&A] [answered]
Hello, community!

Say I have a page written in markdown and rendered using the `react-markdown` package. In this page, I want to programmatically add an arbitrary element halfway through the page but ONLY above headings. I do not want to add anything to my markup. I want this element to be added programmatically.

The arbitrary element is a React Component that has an ad banner.

Thanks in advance!

**Comment by wooorm** [comment, marked as answer]
Write a plugin, walk the tree, find a heading you like, add it

**Reply by ChristianMurphy** [reply]
Adding on, to learn more about plugins, see: https://unifiedjs.com/learn/
For inspiration around adding things in/near to headings, checkout: https://github.com/rehypejs/rehype-autolink-headings

**Reply by iamhectorsosa** [reply]
Thanks, guys. Will check it out. @wooorm @ChristianMurphy 

My project has a really old version of the package so I ended using the root parser, filtering my headings and then appending the component to the selected heading. A bit of a dirty solution but it works.

Seems that a problem like this would have a solution already but will take a look at the information to create a plugin to address this. Thanks!
